src/components/synapse.rs

```diff
@@ -45,22 +54,52 @@ impl SynapseComponent {
         result
     }
 
-    pub async fn who_am_i(&self, token: &str) -> Result<WhoAmIResponse, String> {
+    pub async fn who_am_i(&self, token: &str) -> Result<WhoAmIResponse, CommonError> {
```

Comment:
cool! why not implement it for `get_version`?  just to have the same contract

Reply:
Implemented!